src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -63,6 +64,48 @@ public TermsOfUseHelper(TermsOfUseHelperParams touParams)
             authenticationManager = touParams.AuthenticationManager;
         }
 
+        public void ExecuteForTou()
+        {
+            if (Interlocked.Increment(ref lockCount) > 1)
+            {
+                // Determine if there is already a previous request to display
+                // terms of use dialog, if so, do nothing for the second time.
+                Interlocked.Decrement(ref lockCount);
+                return;
+            }
+
+            Task<bool>.Factory.StartNew(() => packageManagerClient.GetTermsOfUseAcceptanceStatus()).
+                ContinueWith(t =>
+                {
+                    // The above GetTermsOfUseAcceptanceStatus call will get the
+                    // user to sign-in. If the user cancels that sign-in dialog 
+                    // without signing in, we won't show the terms of use dialog,
+                    // simply return from here.
+                    // 
+                    if (authenticationManager.LoginState != LoginState.LoggedIn)
+                        return;
+
+                    var termsOfUseAccepted = t.Result;
+                    if (termsOfUseAccepted)
+                    {
+                        // Terms of use accepted, proceed to publish.
+                        callbackAction.Invoke();
+                    }
+                    else
+                    {
+                        // Prompt user to accept the terms of use.
+                        ShowTermsOfUseForPublishing();
+                    }
+
+                }, TaskScheduler.FromCurrentSynchronizationContext()).
+                ContinueWith(t =>
+                {
+                    // Done with terms of use dialog, decrement counter.
+                    Interlocked.Decrement(ref lockCount);
+
+                }, TaskScheduler.FromCurrentSynchronizationContext());
+        }
+
         public void Execute()
         {
             if (Interlocked.Increment(ref currentRequestCount) > 1)
```

Comment:
I do not see any difference between `Execute` and `ExecuteForTou`, would it be nicer if we update the existing `Execute` to take an additional parameter like so:

```csharp
public void Execute(bool preventReentrant)
{
    if (preventReentrant && (Interlocked.Increment(ref lockCount) > 1)) {
        // Existing code goes here...
    }
}
```

And then for the caller below, simply pass `true` to it.